### ISSUE TITLE
PPPConnection not unblocking

### ISSUE THREAD

**Issue by andreibosco** [COUR4G3/python-pppd#1]
I'm having a strange issue with PPPConnection. I've installed `python-pppd` using `pip` and tried to follow the _getting started_ steps, but the PPPConnection command doesn't unblock after connecting. The ppp connection is successful, I am able to ping and browse but looks like PPPConnection doesn't recognized it as successful and stays stuck. If I cancel it using `ctrl+C` the ppp connection is terminated.

Any idea of what might be happening? By the way, this happens with python 2 and 3.

I'm attaching a photo showing the output of `tail -f /var/log/syslog/`, `ping www.google.com` and the python console

![screenshot-raspberry](https://cloud.githubusercontent.com/assets/607297/20735908/03184ac4-b682-11e6-8cba-e4f24ee424bf.jpg)
.

**Comment by COUR4G3** [comment]
Was running into a similar issue when using the .connected property, the subprocess stdout read is a blocking read, have since changed it to a non-blocking one and uploaded the new version. Check and see if this works for you.

**Comment by andreibosco** [comment]
Awesome, it works now, but I might have found another issue: `ppp.disconnect()` doesn't work. It returns the following:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pppd.py", line 138, in disconnect
    self.proc.send_signal(signal.SIGHUP)
NameError: global name 'signal' is not defined
```

**Update:**
I've found how to fix this new issue. I will close this issue, open a new one and make a pull request, to keep things organized.
